kore/src/SMT/SimpleSMT.hs

```diff
@@ -249,33 +253,32 @@ data Value =  Bool  !Bool           -- ^ Boolean value
 
 --------------------------------------------------------------------------------
 
-type Logger = Log.LogAction IO Log.SomeEntry
-
 -- | An interactive solver process.
 data Solver = Solver
     { hIn    :: !Handle
     , hOut   :: !Handle
     , hErr   :: !Handle
     , hProc  :: !ProcessHandle
-    , logger :: !Logger
```

Comment:
We can keep this module in `IO` by introducing a new data type `SolverHandle` with just the four handle fields. Every function that returns a `Solver` now will return a `SolverHandle` instead, and nothing else in this module needs to change. In `SMT`, we can keep a `SolverHandle` instead of a `Solver`, and assemble the `Solver` from the `LogAction` and the `SolverHandle` when we call a function that needs it.